File: core/src/main/java/com/core/abstractbase/AbstractView.java
package com.core.abstractbase;

import android.content.Context;

/**
 * @author volkanhotur
 */

public interface AbstractView {

    void showLoadingBar();

    void hideLoadingBar();

    boolean isLoadingBarEnabled();

    void expireSession();

    void navigateView();

    Context context();
}

File: core/src/main/java/com/core/abstractbase/AbstractView.kt
package com.core.abstractbase

import android.content.Context

/**
 * @author volkanhotur
 */
interface AbstractView {

    fun showLoadingBar()

    fun hideLoadingBar()

    val isLoadingBarEnabled: Boolean

    fun expireSession(errorTitle : String?, errorMessage: String?)

    fun navigateView()

    fun context(): Context?
}